projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/notebook_plugin.py

```diff
@@ -22,6 +26,22 @@ def initialize_job(self, context: JobContext):
             for file_path in notebook_files:
                 Notebook.register_notebook_steps(file_path, context)
 
+    @hookimpl(hookwrapper=True)
+    def run_job(self, context: JobContext) -> None:
+        out: HookCallResult
+        out = yield
+        result: ExecutionResult = out.get_result()
+        step_results = result.__dict__["steps_list"]
+        for step_result in step_results:
+            if step_result.__dict__["status"] == ExecutionStatus.ERROR:
+                error_info = {
+                    "step_name": step_result.__dict__["name"],
+                    "blamee": step_result.__dict__["blamee"].__str__(),
+                    "details": step_result.__dict__["details"],
+                }
+                with open("error.json", "w") as outfile:
```

Comment:
do not hard code the file name or path and certainly not based on relative path.

 This could end up anywhere. Have a well known location (e.g home dir (~/.vdk/error/job-name/ for example) that can be overridden by configuration if needed.

Reply:
Wouldn't it be better to have it in the job directory?

Reply:
Inside the job directly - probably not or it can be deployed alongside the job accidentally. 

That's why the keytab is not being kept there. 

But it could be one level up . If we have a job directory "/home/jobs/xxx-job/ 
Maybe we can keep it at /home/jobs/.error.json 

Also make it hidden file (with a dot) by default. 


But the issue is that the current code will save it in the current working directory which can change easily depending on context and how the program was started. We need more predictable and fixed place.

Reply:
I moved it to the parents directory and made it hidden